internal/pkg/pubsub/pubsub.go

```diff
@@ -0,0 +1,239 @@
+package pubsub
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/internal/pkg/pubsub/google"
+	"github.com/frain-dev/convoy/internal/pkg/pubsub/sqs"
+	"github.com/frain-dev/convoy/pkg/httpheader"
+	"github.com/frain-dev/convoy/pkg/log"
+	"github.com/frain-dev/convoy/queue"
+	"github.com/frain-dev/convoy/util"
+	"github.com/frain-dev/convoy/worker/task"
+	"github.com/google/uuid"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type PubSub interface {
+	Start()
+	Consume()
+	Stop()
+}
+
+type Source struct {
+	// The pub sub client.
+	client PubSub
+
+	// This is an identifier for the source config used to
+	// track if an existing source config has been changed.
+	hash string
+}
+
+type SourcePool struct {
```

Comment:
@Dotunj I think we can make this easier to understand - `SourcePool` handles the storage of sources. `SourceLoader` handles refreshing from the storage. 

I believe it'll make the APIs easier to reason about. The `SourcePool` will be a wrapper around maps for our own logic of `insertion`, `deletion`, `count` etc. At the same time, `SourceLoader` will be responsible for running at intervals to load new data from the backing store. 

What do you think?

Reply:
One example where it is confusing to reason about is two insert methods. :(